### PR TITLE
Added toUnFoldable function for Data.Maybe

### PR DESCRIPTION
**Pull request by eskimor**
This needs to go to the unfoldable package instead of the maybe package
to avoid a circular dependency.

### CODE REVIEW

File: src/Data/Unfoldable/Maybe.purs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Data.Unfoldable.Maybe where
+
+import Data.Maybe (Maybe(Nothing))
+import Data.Tuple (Tuple(Tuple))
+import Data.Unfoldable (unfoldr, class Unfoldable)
+import Prelude ((<$>), flip)
+
+
+-- | Convert a Maybe to any Unfoldable like lists and arrays.
+toUnfoldable :: forall f a. Unfoldable f => Maybe a -> f a
```
Comment (paf31) [inline src/Data/Unfoldable/Maybe.purs]: I would put this into `Data.Unfoldable` and call it `fromMaybe`.

Comment (eskimor) [inline src/Data/Unfoldable/Maybe.purs]: Fine with me! - I just chose toUnfoldable for consistency with other libraries.

**Comment by eskimor** [comment]
Done.

**Comment by eskimor** [comment]
is it ok now?

**Comment by paf31** [comment]
Yes, thanks!

**Pull request merged**